### PR TITLE
emulator: Add signing process effect

### PR DESCRIPTION
**Pull request by j-mueller**
* Add a multi-agent effect for signing transactions to the emulator
* Remove the `sign` function from `WalletAPI`

### CODE REVIEW

File: plutus-contract/src/Language/Plutus/Contract/Wallet.hs [hunk 1]
```diff
@@ -166,28 +162,6 @@ addOutputs pk vl tx = tx & over Tx.outputs (pko :) where
...
-
--- | Sign the transaction with the private keys of the given wallets,
---   ignoring the list of public keys that the 'SigningProcess' is passed.
-signWallets :: [Wallet] -> SigningProcess
-signWallets wallets = SigningProcess $ \_ tx ->
-    let signingKeys = E.walletPrivKey <$> wallets in
-    pure (foldr Tx.addSignature tx signingKeys)
+handleTx :: (MonadWallet m, ChainIndexAPI m, SigningProcessAPI m) => UnbalancedTx -> m Tx
+handleTx utx =
+    balanceWallet utx >>= addSignatures (Set.toList $ unBalancedTxRequiredSignatories utx) >>= WAPI.signTxAndSubmit
```
Comment (raduom) [inline plutus-contract/src/Language/Plutus/Contract/Wallet.hs]: Is this a typo?

Comment (j-mueller) [reply plutus-contract/src/Language/Plutus/Contract/Wallet.hs]: What do you mean is the typo?

Comment (raduom) [reply plutus-contract/src/Language/Plutus/Contract/Wallet.hs]: I meant the `unBalancedTxRequiredSignatories` function.

File: plutus-emulator/src/Wallet/Emulator/SigningProcess.hs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    let signingKeys = Wallet.walletPrivKey <$> wallets in
+    pure (foldr Tx.addSignature tx signingKeys)
+
+instance Show SigningProcess where
+    show = const "SigningProcess <...>"
+
+data SigningProcessEffect r where
+    AddSignatures :: [L.PubKeyHash] -> Tx -> SigningProcessEffect Tx
+    SetSigningProcess :: SigningProcess -> SigningProcessEffect ()
+makeEffect ''SigningProcessEffect
```
[the remark this answers is not in the document]
Comment (j-mueller) [reply plutus-emulator/src/Wallet/Emulator/SigningProcess.hs]: Yeah you are right, maybe there should be a separate `EmulatorControl` effect for that kind of stuff.

**Pull request merged**